x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/JwtRealm.java

```diff
@@ -43,71 +42,84 @@
 import org.elasticsearch.xpack.security.authc.support.ClaimParser;
 import org.elasticsearch.xpack.security.authc.support.DelegatedAuthorizationSupport;
 
-import java.net.URL;
+import java.io.IOException;
+import java.net.URI;
 import java.nio.charset.StandardCharsets;
-import java.nio.file.Files;
-import java.nio.file.Path;
 import java.util.Arrays;
-import java.util.Collections;
+import java.util.LinkedHashMap;
 import java.util.List;
-import java.util.Locale;
 import java.util.Map;
 import java.util.TreeSet;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
 
 public class JwtRealm extends Realm implements CachingRealm, Releasable {
-
     private static final Logger LOGGER = LogManager.getLogger(JwtRealm.class);
 
```

Comment:
I like this type of refactoring, but I usually leave it to the end of a feature. Prematurely removing fields that I need to re-add later just increases work. Can we leave that to the end of the feature?

Reply:
I see this the other way: Prematurely adding everything as a field in case you need them at some point just makes it more work to remove them later :) 

This is the PR that I'm reviewing so this is my chance to make this comment. I'm happy for you to add a TODO here and  defer this to a later PR

Reply:
Removed a lot of the JwtRealm members. The number of members is much smaller now.